api/src/test/java/org/openmrs/test/BaseContextSensitiveTest.java

```diff
@@ -858,6 +858,7 @@ protected IDatabaseConnection setupDatabaseConnection(Connection connection) thr
 	public synchronized void deleteAllData() {
 		try {
 			Context.clearSession();
```

Comment:
Since we are calling this together with clear session, what do you think about simply putting it in clearSession()?

Reply:
Well clearSession only clears the DB session (calls getCurrentSession().clear()). It doesn't even clear Hibernate second level cache. It would be a change to the behaviour of clearSession. There's clearEntireCache, which used to clear only Hibernate second level cache so I added clearing api cache there. We could have some other method like clearSessionWithCache to do it all in one go.